### PR TITLE
Add test distros

### PR DESCRIPTION
**Pull request by rhdxmr**
- Run build test with `foniod-build` images
- Add ubuntu 18.04 containing kernel v4.19 headers
- Add ubuntu 21.04 and alpine 3.14
- Add `scripts/build-test.sh`

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -33,32 +59,79 @@ jobs:
...
+        /bin/bash -c 'make -C /lib/modules/*/build prepare; cargo clean && export KERNEL_VERSION=$(ls --indicator-style=none /lib/modules/) && echo KERNEL_VERSION=$KERNEL_VERSION && cargo build && cargo build --examples'
+    - name: Run RedBPF build with vmlinux on fedora-34 container
+      run: |
+        docker run --privileged \
+        -v $PWD:/build \
+        -w /build \
+        $BASE_IMAGE:latest-fedora-34 \
+        /bin/bash -c 'cargo clean && /lib/modules/*/build/scripts/extract-vmlinux /lib/modules/*/vmlinuz > /boot/vmlinux && export REDBPF_VMLINUX=/boot/vmlinux && cargo build && cargo build --examples'
 
-  fedora-35-build-test:
```
Comment (rsdy) [inline .github/workflows/build-test.yml:38]: Why don't we do Fedora 35 builds anymore?

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -33,32 +59,79 @@ jobs:
...
-        $BASE_IMAGE:latest-fedora-rawhide \
-        /bin/bash -c 'export KERNEL_VERSION=$(ls --indicator-style=none /lib/modules/); echo KERNEL_VERSION=$KERNEL_VERSION; cargo build && cargo build --examples'
+        $BASE_IMAGE:latest-alpine \
+        sh -c 'cargo clean && export KERNEL_VERSION=$(ls --indicator-style=none /lib/modules/) RUSTFLAGS=-Ctarget-feature=-crt-static && echo KERNEL_VERSION=$KERNEL_VERSION && cargo +1.51 build --no-default-features --features llvm11 && cargo +1.51 build --no-default-features --features llvm11 --examples'
 
-  publish: 
+  publish:
     runs-on: ubuntu-latest
     if: startsWith( github.ref, 'refs/tags/v')
     needs:
```
Comment (rsdy) [inline .github/workflows/build-test.yml:133]: Please add all the distros into this list, as this is the list of required checks before a new release.

**Review by rsdy — CHANGES_REQUESTED** [review]

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (rhdxmr) [reply .github/workflows/build-test.yml:38]: fedora 35 is not released yet. it is in the development phase.

File: .github/workflows/build-test.yml (hunk 2 above)
Comment (rhdxmr) [reply .github/workflows/build-test.yml:133]: OK

**Review by rsdy — APPROVED** [review]

**Pull request merged**